app/api/items/save_grade.go

```diff
@@ -108,31 +108,21 @@ func saveGradingResultsIntoDB(store *database.DataStore, user *database.User,
 	if validated {
 		// Item was validated
 		columnsToUpdate = append(columnsToUpdate,
-			"sAncestorsComputationState", "bValidated", "bKeyObtained", "sValidationDate",
+			"sAncestorsComputationState", "bValidated", "sValidationDate",
 		)
 		values = append(values,
-			todo, 1, 1, gorm.Expr("IFNULL(sValidationDate, NOW())"))
-		// if the item is validated, we don't need to check whether we are above iScoreMinUnlock
-		// as we are sure we get a higher score than iScoreMinUnlock
+			todo, 1, gorm.Expr("IFNULL(sValidationDate, NOW())"))
+	}
+	if shouldUnlockItems(store, requestData.TaskToken.Converted.LocalItemID, score, validated) {
```

Comment:
In my version, I created another variable than "validated" for the getFullScore arg. Actually, "validated" does not imply "get-full-score", rather the opposite, so it should something based on the score (so `score == 100`). Or, to prevent repeating the max, either put the max as const, or (a few lines earlier):
```
fullScore = score == 100
validated = fullScore 
```

Reply:
fixed